tests/core/string/test_string.h

```diff
@@ -152,6 +152,20 @@ TEST_CASE("[String] UTF16 with BOM") {
 	CHECK(String::utf16(cs) == s);
 }
 
+TEST_CASE("[String] UTF8 with CR") {
+	const String base = U"Hello darkness\r\nMy old friend\nI've come to talk\rWith you again";
+
+	String keep_cr;
+	Error err = keep_cr.parse_utf8(base.utf8().get_data());
+	CHECK(err == OK);
+	CHECK(keep_cr == base);
+
+	String no_cr;
+	err = no_cr.parse_utf8(base.utf8().get_data(), true); // Skip CR.
```

Comment:
```suggestion
	err = no_cr.parse_utf8(base.utf8().get_data(), -1, true); // Skip CR.
```

Second argument is length (or -1 for null terminated).

Reply:
Oh right, thanks. I wish `-Werror` would flag this but it doesn't :(

Reply:
Why would it? `true` to `1` conversion is completely valid (at least in C++11 and newer).

Reply:
It's valid, but I wish it would flag it as a warning. It's often a mistake and not intentional.